cmd/kubeadm/app/componentconfigs/configset.go

```diff
@@ -69,11 +69,11 @@ func (h *handler) FromDocumentMap(docmap kubeadmapi.DocumentMap) (kubeadmapi.Com
 
 // fromConfigMap is an utility function, which will load the value of a key of a config map and use h.FromDocumentMap() to perform the parsing
 // This is an utility func. Used by the component config support implementations. Don't use it outside of that context.
-func (h *handler) fromConfigMap(client clientset.Interface, cmName, cmKey string, mustExist bool) (kubeadmapi.ComponentConfig, error) {
+func (h *handler) fromConfigMap(client clientset.Interface, cmName, cmKey string) (kubeadmapi.ComponentConfig, error) {
 	configMap, err := apiclient.GetConfigMapWithRetry(client, metav1.NamespaceSystem, cmName)
 	if err != nil {
-		if !mustExist && (apierrors.IsNotFound(err) || apierrors.IsForbidden(err)) {
-			klog.Warningf("Warning: No %s config is loaded. Continuing without it: %v", h.GroupVersion, err)
+		if apierrors.IsNotFound(err) || apierrors.IsForbidden(err) {
+			klog.Infof("No %s config map found. Either an addon is not installed, or this is a generated config for a core component", h.GroupVersion)
```

Comment:
Is it right to use `klog.Infof` here?
If I remember well, for the user-facing stuff we are using `fmt.Printl` and the `[phase]` notation, while for the debug information we are using `klog.V(x).Infof`

Reply:
Also the part "Either an addon is not installed" is a little bit confusing. What is the use case an addon is not installed but we are trying to read the corresponding config map. It should not be possible to avoid to read the config map if an addon is not installed

Reply:
yes, we don't use `klog.Infof()`, it has to be `klog.Warningf`, but probably best to use `klog.V(1).Infof()` for this "warning".

Reply:
> Also the part "Either an addon is not installed" is a little bit confusing. What is the use case an addon is not installed but we are trying to read the corresponding config map. It should not be possible to avoid to read the config map if an addon is not installed

Currently we do that for the kube-proxy when it's phase was skipped. We always try to load its config map, but we ignore the error if it's missing.
Also, if this implementation is to be used, the "warning" needs to be downgraded a bit to an info (I like the `V(1)` suggestion), because kubeadm generated configs won't be in config maps. Again, with the checksum implementation we can keep things as they are here.